erasure-readfile.go

```diff
@@ -74,13 +67,33 @@ func erasureReadFile(disks []StorageAPI, volume string, path string, partName st
 			if disk == nil {
 				continue
 			}
+
 			// Initialize shard slice and fill the data from each parts.
-			enBlocks[blockIndex] = make([]byte, curEncBlockSize)
+			enBlocks[blockIndex] = make([]byte, eInfo.BlockSize)
+
 			// Read the necessary blocks.
-			_, err := disk.ReadFile(volume, path, offsetEncOffset, enBlocks[blockIndex])
+			n, err := disk.ReadFile(volume, path, block*eInfo.BlockSize, enBlocks[blockIndex])
 			if err != nil {
 				enBlocks[blockIndex] = nil
+			} else if n < eInfo.BlockSize {
+				// As the data we got is smaller than eInfo.BlockSize, keep only required shard slice
+				enBlocks[blockIndex] = append([]byte{}, enBlocks[blockIndex][:n]...)
+			}
+
+			// Remember bytes read at first time.
+			if lastReadSize == 0 {
+				lastReadSize = n
+			}
+
+			// TODO: In a suitation, there may be very few corrupted shards among total shards
```

Comment:
can you open a bug on this and explain the situation there? - need to see how this plays out, since a corrupted block would have been skipped by now after the 'isValidBlock'